specification/machinelearning/Azure.AI.ChatProtocol/operations.tsp

```diff
@@ -0,0 +1,35 @@
+import "@typespec/http";
+
+import "@azure-tools/typespec-azure-core";
+
+import "./models/blocking.tsp";
+import "./models/streaming.tsp";
+
+namespace Azure.AI.ChatProtocol;
+
+using TypeSpec.Http;
+using Azure.Core.Foundations;
+
+@route("/chat")
+interface Chat<StateModel, ContextModel> {
+  /* Using the Foundations.Operation template so the API version parameter is added */
+  #suppress "@azure-tools/typespec-azure-core/use-standard-operations" "Defining the operation in an Azure independent fashion"
+  @doc("Creates a new streaming chat completion.")
+  @sharedRoute
+  createStreaming is Operation<
+    StreamingChatCompletionOptions<StateModel, ContextModel>,
+    ChatCompletionChunk<StateModel, ContextModel>
+  >;
+
+  /* Using the Foundations.Operation template so the API version parameter is added */
+  #suppress "@azure-tools/typespec-azure-core/use-standard-operations" "Defining the operation in an Azure independent fashion"
+  @doc("Creates a new chat completion.")
+  @sharedRoute
+  create is Operation<
+    ChatCompletionOptions<StateModel, ContextModel>,
+    ChatCompletion<StateModel, ContextModel>
+  >;
+}
+
+@route("/chat")
+interface GenericChatClient extends Chat<unknown, Record<unknown>> {}
```

Comment:
Me too is not conformable with `unknown` as both `StateModel` and `ContextModel` (though you may have it for SDK concern).

Reply:
Unfortunately, given the way the services that will implement this API behave, we can't really make the parameters have a stricter schema. It is possible that, in the future, we could extend the client libraries to provide some level of structure for common use cases, but it is still a bit early to do that. In some sense, this is similar to what we had to do to support functions in openai, and the guidance there was to allow for customers to bring their own json.

Reply:
To expand on this, we expect customers to "subclass" the service and thus we want to give them the flexibility to better type their particular use. All such protocols will be viewed as adhering to the base Chat API protocol.